webserver/webserver/src/main/java/io/helidon/webserver/UpgradeCodecSupplier.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.webserver;
+
+import java.util.Optional;
+
+import io.netty.handler.codec.ByteToMessageDecoder;
```

Comment:
Using netty in API/SPI has been discouraged in Helidon and should not be used. We need to decide whether this case is strong enough to bypass this rule

Reply:
abstracted away